fferr/providers.go

```diff
@@ -22,14 +22,27 @@ type ConnectionError struct {
 	baseGRPCError
 }
 
-func NewExecutionError(providerName, resourceName, resourceVariant, resourceType string, err error) *ExecutionError {
+func NewExecutionError(providerName string, err error) *ExecutionError {
```

Comment:
What do we think about this being a ProviderError and just having the other one be ExecutionError?

Reply:
We could do that. I thought about that, but waffled given they're both provider errors.

Reply:
Just adding the resolution from our conversation: we'll keeping the naming as is for now and reconsider at the point when we're ready to merge the base branch and the whole team can weigh in.